src/axislike.rs

```diff
@@ -688,3 +703,22 @@ impl From<DualAxisData> for Vec2 {
         data.xy
     }
 }
+
+/// The shape of the deadzone for a [`DualAxis`] input.
+/// 
+/// Uses the x and y thresholds on [`DualAxis`] to set the shape diamensions
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum DeadzoneShape{
```

Comment:
IMO we should merge the deadzone properties into this enum. That should improve clarity, and enable more exotic deadzone shapes with more or less parameters in the future.

Reply:
I initially did do this like:
```rust
pub enum DeadzoneShape{
    Cross(f32, f32),
}
```
but felt that `SingleAxis` thresholds then served no purpose, so went for using the thresholds. But, I agree with merging in the properties now.

Would the merge look like:
```rust
// 1
pub enum DeadzoneShape{
    Cross(f32, f32),
}

// 2
pub enum DeadzoneShape{
    Cross {x: f32, y: f32},
}

// or 3?
pub enum DeadzoneShape{
    Cross(CrossShapeProperties),
}

pub struct CrossShapeProperties{
    x: f32,
    y: f32
}
```

- 1 is the fine lacks documentation on what the values mean unless I placed the docs at `Cross`
- 2 is more clear than 1 but might be too much
- 3 seems like way too much but is an option and might be the best

Reply:
2 is my preference: named fields are really nice.